src/main/java/lotto/LottoResult.java

```diff
@@ -0,0 +1,64 @@
+package lotto;
+
+import static lotto.constants.CalculationConstants.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lotto.view.ConsoleOutput;
+
+public class LottoResult {
+    private final Map<Rank, Integer> rankCounts;
+
+    public LottoResult(List<Lotto> lottos, WinningLotto winningLotto) {
+        this.rankCounts = calculateRankCounts(lottos, winningLotto);
+    }
+
+    private Map<Rank, Integer> calculateRankCounts(List<Lotto> lottos, WinningLotto winningLotto) {
+        Map<Rank, Integer> counts = initializeRankCounts();
+
+        for (Lotto lotto : lottos) {
+            Rank rank = winningLotto.match(lotto);
+            counts.put(rank, counts.get(rank) + 1);
+        }
+
+        return counts;
+    }
+
+    private Map<Rank, Integer> initializeRankCounts() {
+        Map<Rank, Integer> counts = new HashMap<>();
+        for (Rank rank : Rank.values()) {
+            counts.put(rank, 0);
+        }
+        return counts;
+    }
+
+    public long calculateTotalPrize() {
+        long total = 0;
+        for (Rank rank : rankCounts.keySet()) {
+            total += (long) rank.getPrize() * rankCounts.get(rank);
+        }
+        return total;
+    }
+
+    public double calculateProfitRate(int purchaseAmount) {
+        double rate = (double) calculateTotalPrize() / purchaseAmount * PERCENTAGE_MULTIPLIER;
+        return Math.round(rate * PROFIT_RATE_SCALE) / PROFIT_RATE_SCALE;
+    }
+
+    public void printStatistics(ConsoleOutput output) {
+        printRank(output, Rank.FIFTH);
+        printRank(output, Rank.FOURTH);
+        printRank(output, Rank.THIRD);
+        printRank(output, Rank.SECOND);
+        printRank(output, Rank.FIRST);
+    }

```

Comment:
stream 사용하시면 좋을 것 같아요!
```suggestion
public void printStatistics(ConsoleOutput output) {
    Arrays.stream(Rank.values())
          .sorted(Comparator.comparingInt(Rank::getOrder))
          .forEach(rank -> printRank(output, rank));
}
```